signature.c

```diff
@@ -603,7 +605,25 @@ static int pv_signature_parse_certs(struct dl_list *certs_raw, struct mbedtls_x5
 	return ret;
 }
 
-static int pv_signature_load_trust_certs(const char *path, struct mbedtls_x509_crt *cacerts)
+static int _set_path_trust_crts(struct mbedtls_x509_crt *certs,
+				const char *path)
+{
+	config_index_t name = PV_SECUREBOOT_TRUSTSTORE;
+
+	char oem_name[256];
+	SNPRINTF_WTRUNC(oem_name, 256, "oem-%s",
+			pv_config_get_str(PV_SECUREBOOT_OEM_ID));
+
+	// subject CN is taken from the first certificate of the chain
+	if (pv_str_matches(oem_name, strlen(oem_name), certs->subject.val.p,
```

Comment:
is .p the parsed CN= field of subject? or the whole Subject? if not we need to get the CN= field from that subject and compare that one i believe

Reply:
seems like subject is a list, so we have to check the "CN" in subject.oid and then get the value from subject.val